dropwizard-zipkin/src/main/java/smartthings/dw/zipkin/SqsSpanReporterFactory.java

```diff
@@ -0,0 +1,85 @@
+package smartthings.dw.zipkin;
+
+import com.amazonaws.auth.*;
+import com.amazonaws.services.securitytoken.AWSSecurityTokenServiceClientBuilder;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import org.hibernate.validator.constraints.NotEmpty;
+import zipkin.Span;
+import zipkin.reporter.AsyncReporter;
+import zipkin.reporter.Reporter;
+import zipkin.reporter.sqs.SQSSender;
+
+@JsonTypeName("sqs")
+public class SqsSpanReporterFactory implements SpanReporterFactory {
+
+    @NotEmpty
+    private String queueUrl;
+
+    private String awsAccessKeyId;
+
+    private String awsSecretAccessKey;
+
+    private String awsRoleArn;
+
+    @JsonProperty
+    public String getQueueUrl() {
+        return queueUrl;
+    }
+
+    @JsonProperty
+    public void setQueueUrl(String queueUrl) {
+        this.queueUrl = queueUrl;
+    }
+
+    @JsonProperty
+    public String getAwsAccessKeyId() {
+        return awsAccessKeyId;
+    }
+
+    @JsonProperty
+    public void setAwsAccessKeyId(String awsAccessKeyId) {
+        this.awsAccessKeyId = awsAccessKeyId;
+    }
+
+    @JsonProperty
+    public String getAwsSecretAccessKey() {
+        return awsSecretAccessKey;
+    }
+
+    @JsonProperty
+    public void setAwsSecretAccessKey(String awsSecretAccessKey) {
+        this.awsSecretAccessKey = awsSecretAccessKey;
+    }
+
+    @JsonProperty
+    public String getAwsRoleArn() {
+        return awsRoleArn;
+    }
+
+    @JsonProperty
+    public void setAwsRoleArn(String awsRoleArn) {
+        this.awsRoleArn = awsRoleArn;
+    }
+
+    @Override
+    public Reporter<Span> build() {
+        AWSCredentialsProvider provider = (awsAccessKeyId != null && awsSecretAccessKey != null)?
+            new AWSCredentialsProviderChain(
+                new AWSStaticCredentialsProvider(new BasicAWSCredentials(awsAccessKeyId, awsSecretAccessKey))
+            ) : new DefaultAWSCredentialsProviderChain();
+
+        provider = awsRoleArn != null ? new STSAssumeRoleSessionCredentialsProvider
+            .Builder(awsRoleArn, "zipkin")
+            .withStsClient(AWSSecurityTokenServiceClientBuilder.standard().withCredentials(provider).build())
+            .build() :
+            provider;
```

Comment:
Maybe use  if / else if logic instead?

Reads a little confusing to me since we're potentially creating, assigning, and then re-creating re-assigning when it falls through to the end.  Nice just to create and assign to provider once.